### PR TITLE
feat(frontier):add pagination to ListOrganizations

### PR DESCRIPTION
**Pull request by AmanGIT07**
Add new fields to `ListAllOrganizationsRequest` and `ListAllOrganizationsResponse` to enable pagination.

Fields added:
`page_size` ,`page_num` to `ListAllOrganizationsRequest` message
`total_pages` to `ListAllOrganizationsResponse` message

### CODE REVIEW

**Review by anujk14 — CHANGES_REQUESTED** [review]

File: raystack/frontier/v1beta1/admin.proto [hunk 1]
```diff
@@ -453,10 +453,27 @@ message ListGroupsResponse {
...
+      gte: 1,
+      ignore_empty: true,
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The page number to return. The default is 1."}
+  ];
 }
 
 message ListAllOrganizationsResponse {
   repeated Organization organizations = 1;
+  int32 total_pages = 2[
```
[the remark this answers is not in the document]
Comment (anujk14) [reply raystack/frontier/v1beta1/admin.proto:474]: Also, the term `total_pages` is dependant on the `page_size` that the customer has chosen. If we name it as `total_pages` we will need to keep changing it as per the `page_size` in request. `count` would be more standard and uniform, in my opinion.

[the remark this answers is not in the document]
Comment (AmanGIT07) [reply raystack/frontier/v1beta1/admin.proto:474]: `count` gives us total items in the response. I will be `<= page_size` requested. `total_pages` gives us the total number of pages which frontend can use to populate page footers. Both are serving different purposes

[the remark this answers is not in the document]
Comment (anujk14) [reply raystack/frontier/v1beta1/admin.proto:474]: I think number of pages is a fairly straightforward calculation. However, `total_pages` still does not give us the total count of objects, should we want to show that number somewhere.
The API should be fairly independent of the consumer(UI in this case).

[the remark this answers is not in the document]
Comment (AmanGIT07) [reply raystack/frontier/v1beta1/admin.proto]: Updated the `ListAllOrganizationsResponse` message to give count as output

**Review by anujk14 — APPROVED** [review]

**Pull request merged**